File: core/src/nl/t64/game/rpg/screens/world/MiniMapListener.java
package nl.t64.game.rpg.screens.world;

import com.badlogic.gdx.Input;
import com.badlogic.gdx.InputAdapter;
import lombok.AllArgsConstructor;
import nl.t64.game.rpg.constants.Constant;


@AllArgsConstructor
class MiniMapListener extends InputAdapter {

    private final Runnable closeMiniMap;

    @Override
    public boolean keyDown(int keycode) {
        switch (keycode) {
            case Constant.KEYCODE_SELECT,
                    Constant.KEYCODE_RIGHT,
                    Input.Keys.ESCAPE,
                    Input.Keys.M -> closeMiniMap.run();
        }
        return false;
    }

}

File: core/src/nl/t64/game/rpg/screens/world/MiniMapListener.kt
package nl.t64.game.rpg.screens.world

import com.badlogic.gdx.Input
import com.badlogic.gdx.InputAdapter
import nl.t64.game.rpg.constants.Constant


internal class MiniMapListener(
    private val closeMiniMap: () -> Unit
) : InputAdapter() {

    override fun keyDown(keycode: Int): Boolean {
        when (keycode) {
            Constant.KEYCODE_SELECT,
            Constant.KEYCODE_RIGHT,
            Input.Keys.ESCAPE,
            Input.Keys.M -> closeMiniMap.invoke()
        }
        return false
    }

}
